### PR TITLE
MAINT: refactor device offloading

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
refactored `_get_backend` function:
removed unused variable and return value.

### CODE REVIEW

**Review by KulikovNikita — APPROVED** [review]
Looks fine to me

File: sklearnex/_device_offload.py [hunk 1]
```diff
@@ -128,23 +128,23 @@ def _transfer_to_host(queue, *data):
...
 
-    if (cpu_device and obj._onedal_cpu_supported(method_name, *data)) or \
-       (gpu_device and obj._onedal_gpu_supported(method_name, *data)):
-        return 'onedal', queue, cpu_fallback
+    if (cpu_device and onedal_cpu_supported) or \
+       (gpu_device and onedal_gpu_supported):
+        return 'onedal', queue
     if cpu_device:
-        return 'sklearn', None, cpu_fallback
+        return 'sklearn', None
```
Comment (Alexsandruss) [inline sklearnex/_device_offload.py:138]: This change introduce overhead for gpu support check: `_onedal_gpu_supported` was not called before if first condition for cpu is True.

Comment (samir-nasibli) [reply sklearnex/_device_offload.py:138]: Actually, it calls all checks here, but in any way overhead point is correct. Thanks!

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**